CMakeLists.txt

```diff
@@ -50,6 +51,11 @@ if (OLP_SDK_NO_EXCEPTION)
     set(CMAKE_C_FLAGS "-fno-exceptions ${CMAKE_C_FLAGS}")
 endif()
 
+if (OLP_SDK_NO_EXCEPTION AND NOT OLP_SDK_BOOST_THROW_EXCPETION_EXTERNAL)
+    set(CMAKE_CXX_FLAGS "-DOLP_SDK_BOOST_THROW_EXCEPTION ${CMAKE_CXX_FLAGS}")
```

Comment:
Do we need to expose this macro for the user? Maybe better to define it as a `target_compile_definitions` only for the olp-cpp-sdk-core?